python/lsst/meas/algorithms/detection.py

```diff
@@ -366,11 +370,36 @@ def detectFootprints(self, exposure, doSmooth=True, sigma=None, clearMask=True):
 
         fpSets = pipeBase.Struct(positive=None, negative=None)
 
+        if self.config.doTempLocalBackground:
+            # Subtract the local background from the smoothed image.
```

Comment:
We should definitely estimate it before subtracting (otherwise the smoothing messes up the sigma-clipping), and we do.  From there, I don't think it matters when we subtract it, since even this background should have no power on scales as small as the PSF.